plugins/woocommerce/includes/gateways/paypal/class-wc-gateway-paypal-buttons.php

```diff
@@ -11,152 +13,99 @@
 	exit;
 }
 
+use Automattic\WooCommerce\Gateways\PayPal\Buttons as PayPalButtons;
+
 if ( ! class_exists( 'WC_Gateway_Paypal_Request' ) ) {
 	require_once __DIR__ . '/includes/class-wc-gateway-paypal-request.php';
 }
 
 /**
  * Handles PayPal Buttons.
+ *
+ * @deprecated 10.5.0 Use Automattic\WooCommerce\Gateways\PayPal\Buttons instead. This class will be removed in 11.0.0.
  */
 class WC_Gateway_Paypal_Buttons {
 
 	/**
-	 * The option for the client-id.
-	 *
-	 * @var string
-	 */
-	private const CLIENT_ID_OPTION = 'woocommerce_paypal_client_id';
-
-
-	/**
-	 * The gateway instance.
-	 *
-	 * @var WC_Gateway_Paypal
-	 */
-	private $gateway;
-
-	/**
-	 * Whether the gateway should use Orders v2 API.
-	 *
-	 * @var bool
-	 */
-	private $enabled = false;
-
-	/**
-	 * The request instance.
+	 * The delegated buttons instance.
 	 *
-	 * @var WC_Gateway_Paypal_Request
+	 * @var PayPalButtons
 	 */
-	private $request;
+	private $buttons;
 
 	/**
 	 * Constructor.
 	 *
 	 * @param WC_Gateway_Paypal $gateway The gateway instance.
 	 */
 	public function __construct( WC_Gateway_Paypal $gateway ) {
-		$this->gateway = $gateway;
-		$this->request = new WC_Gateway_Paypal_Request( $this->gateway );
-
-		// phpcs:ignore Generic.Commenting.Todo.TaskFound
-		$this->enabled = $this->gateway->should_use_orders_v2() && 'yes' === $this->gateway->get_option( 'paypal_buttons', 'yes' );
+		$this->buttons = new PayPalButtons( $gateway );
 	}
 
 	/**
 	 * Get the options for the PayPal buttons.
 	 *
 	 * @return array
+	 *
+	 * @deprecated 10.5.0 Use Automattic\WooCommerce\Gateways\PayPal\Buttons::get_options() instead. This method will be removed in 11.0.0.
 	 */
 	public function get_options() {
-		$common_options = $this->get_common_options();
-		$options        = array(
-			'partner-attribution-id' => 'Woo_Cart_CoreUpgrade',
-			'page-type'              => $this->get_page_type(),
-		);
-
-		return array_merge( $common_options, $options );
+		return $this->buttons->get_options();
```

Comment:
As suggested in at least one other review, we should use the WordPress `_deprecated_function()` helper to make it super-clear when code is calling the function. This applies here and in the other functions in this class.
```suggestion
		_deprecated_function( __METHOD__, '10.5.0', PayPalButtons::class . '::get_options()' );
		return $this->buttons->get_options();
```